near-contract-standards/src/fungible_token/core.rs

```diff
@@ -100,3 +100,38 @@ pub trait FungibleTokenCore {
     /// Returns the balance of the account. If the account doesn't exist must returns `"0"`.
     fn ft_balance_of(&self, account_id: AccountId) -> U128;
 }
+
+#[cfg(all(not(target_arch = "wasm32"), feature = "unit-testing"))]
+pub mod tests {
+
+    use near_sdk::test_utils::accounts;
+
+    use super::*;
+
+    fn test_transfer_ok(contract: &mut impl FungibleTokenCore) {
+        assert_eq!(contract.ft_balance_of(accounts(2)), U128(1_000_000_000_000_000));
+        assert_eq!(contract.ft_balance_of(accounts(1)), U128(0));
+        let amount = contract.ft_total_supply().0 / 4;
+        contract.ft_transfer(accounts(1), amount.into(), None);
+        assert_eq!(contract.ft_balance_of(accounts(2)).0, contract.ft_total_supply().0 - amount);
+        assert_eq!(contract.ft_balance_of(accounts(1)).0, amount);
+    }
+
+    fn test_transfer_call_ok(contract: &mut impl FungibleTokenCore) {
+        let amount = contract.ft_total_supply().0 / 4;
+        let result = contract.ft_transfer_call(accounts(1), amount.into(), None, "".to_string());
+        match result {
+            PromiseOrValue::Promise(_promise) => {
+                println!("Handle promise here");
+            }
+            _ => panic!("Expected Promise variant"),
+        }
+        assert_eq!(contract.ft_balance_of(accounts(2)).0, contract.ft_total_supply().0 / 2);
+        assert_eq!(contract.ft_balance_of(accounts(1)).0, contract.ft_total_supply().0 / 2);
+    }
+
+    pub fn test(contract: &mut impl FungibleTokenCore) {
+        test_transfer_ok(contract);
+        test_transfer_call_ok(contract);
```

Comment:
We also need negative tests, such as:

1. Attempt to transfer more tokens that sender account has on its balance - should fail and balance remain the same for both accounts
2. Attempt to transfer to a non-registered account - should fail since storage cost is not covered
3. Attempt to transfer from a non-registered account